imagecraft/models/project.py

```diff
@@ -112,3 +121,13 @@ def _providers_base(cls, base: str) -> bases.BaseAlias | None:
         :returns: Always None because imagecraft only supports bare bases.
         """
         return None
+
+    @field_validator("volumes", mode="after")
+    @classmethod
+    def _validate_volumes(
```

Comment:
It seems pydantic it not validating dictionary **keys** even with the `VolumeName` annotation set above, so the next best thing I thought off was to add this validator. I would be interested to know if there is a better way.

Reply:
interesting; maybe the definition of VolumeName needs to be more similar to [ProjectName](https://github.com/canonical/craft-application/blob/main/craft_application/models/constraints.py#L102)

Reply:
Thanks, this is cleaner. The only downside is that I needed a strict match (`re.fullmatch()`) so for now I had to create a more strict version of the `VALID_PARTITION_REGEX` regex.